rotkehlchen/tests/unit/decoders/test_gearbox.py

```diff
@@ -0,0 +1,264 @@
+from functools import wraps
+from typing import TYPE_CHECKING
+from unittest.mock import patch
+
+import pytest
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.chain.evm.decoding.gearbox.constants import CPT_GEARBOX
+from rotkehlchen.chain.evm.decoding.gearbox.gearbox_cache import (
+    get_zapper_tokens,
+    query_gearbox_data,
+    save_gearbox_data_to_cache,
+)
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_DAI, A_ETH, A_USDC
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.evm_event import EvmEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.tests.utils.ethereum import get_decoded_events_of_transaction
+from rotkehlchen.types import (
+    CacheType,
+    ChecksumEvmAddress,
+    Location,
+    TimestampMS,
+    deserialize_evm_tx_hash,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+def mock_get_zapper_tokens(inquirer, pool_data, underlying_token_address):
+    """Mock function for get_zapper_tokens that sorts the result. This is required to make the
+    VCR tests deterministic."""
+    original_get_zapper_tokens = get_zapper_tokens
+
+    @wraps(original_get_zapper_tokens)
+    def wrapper(inquirer, pool_data, underlying_token_address, *args, **kwargs):
+        result = original_get_zapper_tokens(inquirer, pool_data, underlying_token_address, *args, **kwargs)  # noqa: E501
+        if result is not None:
+            sorted_result = sorted(result)
+            return sorted_result
+        return None
+
+    return wrapper(inquirer, pool_data, underlying_token_address)
+
+
+def patch_get_zapper_tokens(func):
+    """Decorator to patch the get_zapper_tokens function with the mock version."""
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        with patch(
+            'rotkehlchen.chain.evm.decoding.gearbox.gearbox_cache.get_zapper_tokens',
+            new=mock_get_zapper_tokens,
+        ):
+            return func(*args, **kwargs)
+    return wrapper
+
+
+@patch_get_zapper_tokens
```

Comment:
why patch entire test and not only the place it's called?

Reply:
Not needed anymore.